Userland/Libraries/LibDebug/DebugSession.cpp

```diff
@@ -440,14 +440,17 @@ void DebugSession::update_loaded_libs()
         if (lib_name == "libgcc_s.so")
             return IterationDecision::Continue;
 
-        if (m_loaded_libraries.contains(lib_name))
+        FlatPtr base_address = entry.as_object().get("address").to_addr();
+        if (m_loaded_libraries.contains(lib_name)) {
+            // We expect the VM regions to be sorted by address.
+            VERIFY(base_address >= m_loaded_libraries.get(lib_name).value()->base_address);
```

Comment:
This sounds a little inefficient, how about:
```suggestion
        if (auto it = m_loaded_libraries.find(lib_name); it != m_loaded_libraries.end()) {
            // We expect the VM regions to be sorted by address.
            VERIFY(base_address >= it->base_address);
```